lib/puppet/provider/package/yum.rb

```diff
@@ -142,13 +142,24 @@ def self.update_command
     'update'
   end
 
+  def makecache_fast
+    command = [command(:cmd), :makecache, :fast]
+    execute(command, :failonfail => false, :combine => false)
+  end
+
   def install
     wanted = @resource[:name]
     error_level = self.class.error_level
     update_command = self.class.update_command
     # If not allowing virtual packages, do a query to ensure a real package exists
     unless @resource.allow_virtual?
-      execute([command(:cmd), '-d', '0', '-e', error_level, '-y', install_options, :list, wanted].compact)
+      command = [command(:cmd), '-d', '0', '-e', error_level, '-y', install_options, :list, wanted].compact
+      output = execute(command, :failonfail => false, :combine => false)
+
+      if output.exitstatus != 0
```

Comment:
One concern I have here is that if `yum list` is going to fail consistently for some reason, we could potentially be running `yum makecache` for every package resource - this would cause a puppet run to be very slow and use a lot of resources just to ultimately fail. Is it possible to more precise about catching the failure we're expecting to see (the package isn't in the cache)?

Reply:
We specifically *removed* our python shim - I think option 1 is the way to go here